crates/evm/src/evm/db.rs

```diff
@@ -86,6 +86,45 @@ impl<'a, C: sov_modules_api::Context> Database for EvmDb<'a, C> {
     }
 }
 
+impl<'a, C: sov_modules_api::Context> DatabaseRef for EvmDb<'a, C> {
```

Comment:
It's not used anywhere and we don't require this trait, so it's a dead code. This implementation requires `&working_set` instead of `&mut working_set` which is nice, however due to the nature of our codegen we always pass `&mut working_set` into our RPC methods like this:

```
                if method.method_signature.asyncness.is_some() {
                    async_methods_count += 1;
                    quote! {
                        #( #docs )*
                        #signature {
                            <#type_name #ty_generics as ::std::default::Default>::default().#method_name(#(#pre_working_set_args,)* &mut Self::get_working_set(self), #(#post_working_set_args),* ).await
                        }
                    }
                } else {
                    quote! {
                        #( #docs )*
                        #signature {
                            <#type_name #ty_generics as ::std::default::Default>::default().#method_name(#(#pre_working_set_args,)* &mut Self::get_working_set(self), #(#post_working_set_args),* )
                        }
                    }
                }
```

So for now `&working_set` brings nothing but additional mutexes inside of ArrayWitness.